File: lib-cli/tmp/server/producer/events/serverevents_userkickoff.rs
use super::{identification, producer::Control, protocol, Context};
use fiber::server;
use std::str::FromStr;

type BroadcastEventsMessage = (Vec<Uuid>, protocol::Events::Message);
type BroadcastEventsUserConnected = (Vec<Uuid>, protocol::Events::UserConnected);

#[allow(unused_variables)]
pub async fn emit<E: std::error::Error, C: server::Control<E> + Send + Clone>(
    event: protocol::ServerEvents::UserKickOff,
    filter: &identification::Filter,
    context: &mut Context,
    control: &Control<E, C>,
) -> Result<(BroadcastEventsMessage, BroadcastEventsUserConnected), String> {
    panic!("Handler for protocol::ServerEvents::UserKickOff isn't implemented");
}

File: lib-cli/tmp/server-ts.backup/src/producer/events/serverevents.userkickoff.ts
import { Filter, Producer, Context, Protocol } from "../implementation/events";
import { Output } from "../implementation/events/serverevents.userkickoff";

export function emit(
	event: Protocol.ServerEvents.UserKickOff,
	filter: Filter,
	context: Context,
	producer: Producer
): Promise<Output> {
	return Promise.reject(
		new Error(`Handler for event "userkickoff" isn't implemented`)
	);
}
